Optional Feature Enchancments: flunkif and strip delimiters from values

vbdevpro opened the pull request:
**Changes:** 

_1._ Added an optional layout field element "flunkif" that accepts a comma separated list of values that, if matched, excludes the entire record from the final output. Also, updated readme.md, engine rules manual documentation and example json speciifcation file. This currently only works with fixed length recrods. Work has started to include variable length layouts as well. 

_2._ Added an optional parameter to the run method (stripDelimiterValues) that strips the delimiter specified by the outputDelimiter from string field values on the input file. Also strip newline and carrige return characters in an attempt to "clean up" input data and limit parsing errors. Updated readme.md to document usage instructions.

vbdevpro committed:
flunkif and stripdelimiter functionality added

1. Added the optional flunkif field layout element to the json layout file that can accept a comma separated list of values and exclude any records with matching values from the final output and produce a count of the total records excluded. 

2. Added the optional stripDelimiterValue parameter to remove any carriage returns, linefeeds or the chosen field delimiter from individual text field values that may cause problems on the final exported output.

vbdevpro committed:
Merge pull request #1 from vbdevpro/vbdevpro-flunkif-patch

flunkif and stripdelimiter functionality added

vbdevpro committed:
Update README.md

updated to include flunkif and stripDelimiterValues examples.

vbdevpro committed:
Update README.md

vbdevpro committed:
Update convert.py

added string datatype check for flunkif functions and delimiter strip function

vbdevpro committed:
Update engine_rules_manual.md

added flunkif info

vbdevpro committed:
Update README.md

vbdevpro committed:
Update README.md

updated document to clarify optional stripDelimiterValues parameter.

vbdevpro committed:
Update README.md

larandvit commented:
Hi @vbdevpro,

Got your PR. I like stripDelimiterValues functionality. Let me some time, I want to run some tests.

Thank you.
Vitaly

vbdevpro commented:
> Hi @vbdevpro,
> 
> Got your PR. I like stripDelimiterValues functionality. Let me some time, I want to run some tests.
> 
> Thank you. Vitaly

OK. Thanks! I hope it adds something to your very useful library,

larandvit reviewed (changes requested):
Hi Chris,
Can you please add stripdelimitervalues parameter to run it in CLI mode? When we run it in command line mode, it always runs as false for the parameter.

parser.add_argument("--stripdelimitervalues",nargs="?", default="yes" if ParameterDefaults.STRIP_DELIMITER_VALUES else "no", choices=["yes","no"], help="strip delimiter values", metavar='yes/no')

Thanks.

vbdevpro commented:
@larandvit Oh, right! Of course. Sorry about that. I will add.

*Done* 2 new commits for the requested change. Let me know it I you would like me to close this pull request open a new one.

vbdevpro committed:
Update convert.py

Added strip delimiter argument and supporting code to command line functionality.

vbdevpro committed:
Update README.md for --stripdelimitervalues

added --stripdelimitervalues info to readme markdown

larandvit commented:
@vbdevpro 
The pull request has been merged.
Thank you!
Vitaly
